File: NetworkingSample/app/src/main/java/com/example/networkingsample/MainActivity.java
package com.example.networkingsample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;


import com.squareup.picasso.Picasso;

import java.util.List;
import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        try {

            HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
            loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

            OkHttpClient okHttpClient = new OkHttpClient.Builder()
                    .addInterceptor(loggingInterceptor)
                    .connectTimeout(30, TimeUnit.SECONDS)
                    .build();

            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl("https://jsonplaceholder.typicode.com/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(okHttpClient)
                    .build();

            APIService service = retrofit.create(APIService.class);

            service.getTodos().enqueue(new Callback<List<Todo>>() {
                @Override
                public void onResponse(Call<List<Todo>> call, Response<List<Todo>> response) {
                    Log.e("Response size: ", response.body().size() + "");
                }

                @Override
                public void onFailure(Call<List<Todo>> call, Throwable t) {
                    Log.e("Error", t.getMessage());
                }
            });


//            service.getTodoById(1).enqueue(new Callback<Todo>() {
//                @Override
//                public void onResponse(Call<Todo> call, Response<Todo> response) {
//                    Log.e("Todo: ", response.body().getTitle());
//                }
//
//                @Override
//                public void onFailure(Call<Todo> call, Throwable t) {
//
//                }
//            });

//            service.getTodosByUserId(2, false).enqueue(new Callback<List<Todo>>() {
//                @Override
//                public void onResponse(Call<List<Todo>> call, Response<List<Todo>> response) {
//                    Log.e("Response size: ", response.body().size() + "");
//                }
//
//                @Override
//                public void onFailure(Call<List<Todo>> call, Throwable t) {
//
//                }
//            });

            service.addTodo(2, "Todo Something", false).enqueue(new Callback<Todo>() {
                @Override
                public void onResponse(Call<Todo> call, Response<Todo> response) {
                    Log.e("Todo: ", response.body().getTitle());
                }

                @Override
                public void onFailure(Call<Todo> call, Throwable t) {

                }
            });


//            service.getTodosFromUrl("https://jsonplaceholder.typicode.com/todos/").enqueue(new Callback<List<Todo>>() {
//                @Override
//                public void onResponse(Call<List<Todo>> call, Response<List<Todo>> response) {
//                    Log.e("Response size: ", response.body().size() + "");
//                }
//
//                @Override
//                public void onFailure(Call<List<Todo>> call, Throwable t) {
//                    Log.e("Error", t.getMessage());
//                }
//            });
        } catch (Exception e) {
            e.printStackTrace();
        }


        ImageView imageView = findViewById(R.id.image);

        Picasso.get()
                .load("https://unsplash.com/photos/5Oe8KFH5998/download")
                .placeholder(R.drawable.placeholder)
                .error(R.drawable.error)
                .centerCrop()
                .resize(300, 300)
                .rotate(90)
                .into(imageView);

    }
}

File: RetrofitApplication/app/src/main/java/com/example/retrofitapplication/MainActivity.kt
package com.example.retrofitapplication

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_main.*
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {

    var list = mutableListOf<Todo>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val loggingInterceptor = HttpLoggingInterceptor()
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY

        val okHttpClient = OkHttpClient.Builder()
            .addInterceptor(loggingInterceptor)
            .connectTimeout(30, TimeUnit.SECONDS)
            .build()

        val retrofit = Retrofit.Builder()
            .baseUrl("https://jsonplaceholder.typicode.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .client(okHttpClient)
            .build()

        val apiService = retrofit.create(ApiService::class.java)


//        apiService.getTodos().enqueue(object : Callback<List<Todo>> {
//            override fun onFailure(call: Call<List<Todo>>, t: Throwable) {
//                Log.e("Error", t.message)
//            }
//
//            override fun onResponse(call: Call<List<Todo>>, response: Response<List<Todo>>) {
//                Log.e("Response size: ", response.body()!!.size.toString() + "")
//                list.addAll(response.body()!!)
//            }
//        })


//        apiService.getTodoById(1).enqueue(object : Callback<Todo> {
//            override fun onFailure(call: Call<Todo>, t: Throwable) {
//                Log.e("Error", t.message)
//            }
//
//            override fun onResponse(call: Call<Todo>, response: Response<Todo>) {
//                Log.e("Todo title: ", response.body()!!.title + "")
//            }
//        })

//        apiService.getTodosByUserId(1, false).enqueue(object : Callback<List<Todo>> {
//            override fun onFailure(call: Call<List<Todo>>, t: Throwable) {
//                Log.e("Error", t.message)
//            }
//
//            override fun onResponse(call: Call<List<Todo>>, response: Response<List<Todo>>) {
//                Log.e("Response size: ", response.body()!!.size.toString() + "")
//            }
//        })


//        apiService.addTodo(2, "Do something", false).enqueue(object : Callback<Todo> {
//            override fun onFailure(call: Call<Todo>, t: Throwable) {
//                Log.e("Error", t.message)
//            }
//
//            override fun onResponse(call: Call<Todo>, response: Response<Todo>) {
//                Log.e("Todo title: ", response.body()!!.title + "")
//            }
//        })


//        apiService.getTodosWithUrl("https://jsonplaceholder.typicode.com/todos/")
//            .enqueue(object : Callback<List<Todo>> {
//                override fun onFailure(call: Call<List<Todo>>, t: Throwable) {
//                    Log.e("Error", t.message)
//                }
//
//                override fun onResponse(call: Call<List<Todo>>, response: Response<List<Todo>>) {
//                    Log.e("Response size: ", response.body()!!.size.toString() + "")
//                    list.addAll(response.body()!!)
//                }
//            })


        Picasso.get()
            .load("https://www.planetware.com/wpimages/2020/02/france-in-pictures-beautiful-places-to-photograph-eiffel-tower.jpg")
            .resize(300, 300)
            .rotate(90f)
            .into(imageView)
    }
}